File: backend/app/schemas/customer_intelligence.py
from decimal import Decimal
from pydantic import BaseModel


class CustomerIntelligenceFeaturesResponse(BaseModel):
    transaction_count: int
    total_transaction_value: Decimal
    average_transaction_value: Decimal
    days_since_last_transaction: int


class CustomerIntelligencePredictionResponse(BaseModel):
    churn_probability: Decimal


class CustomerIntelligenceActionResponse(BaseModel):
    action_type: str
    priority: int
    reason: str


class CustomerIntelligenceResponse(BaseModel):
    features: CustomerIntelligenceFeaturesResponse
    prediction: CustomerIntelligencePredictionResponse
    action: CustomerIntelligenceActionResponse

File: frontend/src/types/customerIntelligence.ts
export type CustomerIntelligence = {
  features: {
    transaction_count: number;
    total_transaction_value: string;
    average_transaction_value: string;
    days_since_last_transaction: number;
  };

  prediction: {
    churn_probability: string;
  };

  action: {
    action_type: string;
    priority: number;
    reason: string;
  };
};